tests/setup_test.go

```diff
@@ -5856,3 +5862,273 @@ func TestCommitStatusDelivery(t *testing.T) {
+			runs, _, err := gwUser01Client.GetProjectRuns(ctx, project.ID, nil, nil, 0, 0, false)
+			if err != nil {
+				return false, nil
+			}
+
+			if len(runs) == 0 {
+				return false, nil
+			}
+			if runs[0].Phase != rstypes.RunPhaseFinished {
+				return false, nil
+			}
+
+			return true, nil
+		})
+
+		runs, _, err := gwUser01Client.GetProjectRuns(ctx, project.ID, nil, nil, 0, 0, false)
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+		if len(runs) != 1 {
+			t.Fatalf("expected %d run got: %d", 1, len(runs))
+		}
+
+		if runs[0].Phase != rstypes.RunPhaseFinished {
+			t.Fatalf("expected run phase %q, got %q", rstypes.RunPhaseFinished, runs[0].Phase)
+		}
+		if runs[0].Result != rstypes.RunResultSuccess {
+			t.Fatalf("expected run result %q, got %q", rstypes.RunResultSuccess, runs[0].Result)
+		}
+
+		_, _, err = gwUser01Client.ProjectRunAction(ctx, project.ID, runs[0].Number, &gwapitypes.RunActionsRequest{
+			ActionType: gwapitypes.RunActionTypeRestart,
+			FromStart:  true,
+		})
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+
+		// test org run actions user member
+
+		_, _, err = gwUser02Client.ProjectRunAction(ctx, project.ID, runs[0].Number, &gwapitypes.RunActionsRequest{
+			ActionType: gwapitypes.RunActionTypeRestart,
+			FromStart:  true,
+		})
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+
+		// test org run actions user not member
+
+		_, _, err = gwAdminClient.CreateUser(ctx, &gwapitypes.CreateUserRequest{UserName: agolaUser03})
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+
+		token, _, err = gwAdminClient.CreateUserToken(ctx, agolaUser03, &gwapitypes.CreateUserTokenRequest{TokenName: "testtoken"})
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+
+		gwUser03Client := gwclient.NewClient(sc.config.Gateway.APIExposedURL, token.Token)
+
+		_, _, err = gwUser03Client.ProjectRunAction(ctx, project.ID, runs[0].Number, &gwapitypes.RunActionsRequest{
+			ActionType: gwapitypes.RunActionTypeRestart,
+			FromStart:  true,
+		})
+		if err == nil {
+			t.Fatalf("expected error %v, got nil err", expectedErr)
+		}
+		if err.Error() != expectedErr {
+			t.Fatalf("expected err %v, got err: %v", expectedErr, err)
+		}
+
+		// test MembersCanPerformRunActions false user member
+
+		_, _, err = gwUser01Client.UpdateProject(ctx, project.ID, &gwapitypes.UpdateProjectRequest{
+			MembersCanPerformRunActions: util.BoolP(false),
+		})
+		if err != nil {
+			t.Fatalf("unexpected err: %v", err)
+		}
+
+		_, _, err = gwUser02Client.ProjectRunAction(ctx, project.ID, runs[0].Number, &gwapitypes.RunActionsRequest{
+			ActionType: gwapitypes.RunActionTypeRestart,
+			FromStart:  true,
+		})
+		if err == nil {
+			t.Fatalf("expected error %v, got nil err", expectedErr)
+		}
+		if err.Error() != expectedErr {
+			t.Fatalf("expected err %v, got err: %v", expectedErr, err)
+		}
+	})
+
+	t.Run("test user run actions", func(t *testing.T) {
+		dir := t.TempDir()
```

Comment:
What is the behavior when the project owner is an user? I think it should be defined and tested here.